### DISCUSSION TITLE
How to make a plugin of a plugin

### DISCUSSION THREAD

**Discussion by jamesellis1999** [pytest-dev/pytest#10370] [General]
Hi,

I am looking to make a simple plugin to extend the `pytest-postgresql` plugin. One of the default fixtures `pytest-postgresql` provides is called `postgresql_proc`. I want to extend this fixture with another fixture. In essence, my `plugin.py` file looks as such: 

```
# plugin.py

import pytest

@pytest.fixture
def run_migrations(postgresql_proc)
  '''Run migrations on the database'''
   # some alembic migrations code 
```
When I run `pytester` I get the following: 

```
fixture 'postgresql_proc' not found
```

I have read the user manual which suggests the following when collaborating with other plugins (section 2.15.7): 

```
plugin = config.pluginmanager.get_plugin("name_of_plugin")
```
However, it is not clear in the user manual where this should be used. I have tried using it in hooks and within the `run_migrations` fixture (using `request.config.pluginmanager`) but this did not work. 

My setup.cfg looks as such:

```
[options]
zip_safe = False
include_package_data = True
python_requires = >= 3.7
packages = find:
install_requires =
    pytest-postgresql

[options.entry_points]
pytest11 =
    pytest_postgresql_session = pytest_postgresql_session.plugin

[tool:pytest]
testpaths = tests
asyncio_mode = auto
```

Any suggestions would be much appreciated. Thanks!

**Comment by RonnyPfannschmidt** [comment]
Based on the information here it's not clear what is missing, was your plugin installed, did you install the extra plugins you configured in setup.cfg?

**Reply by jamesellis1999** [reply]
Both my plugin and the `pytest-postgresql` are installed via pip on my virtual environment.

**Reply by RonnyPfannschmidt** [reply]
based on the documentation of the plugin, installation should enable/allow it, did you install one of the postgresql client libraries mentioned, it may be possible that a import error is hidden by a ux bug

is the fixture available outside of pytester?

**Reply by jamesellis1999** [reply]
The issue turned out to be `pip install` with the `-e` flag. Installing with `-e` flag made the plugin undiscoverable by pytest and pytester for some reason.
